pkg/cluster/internal/providers/podman/provider.go

```diff
@@ -359,44 +358,50 @@ func (p *provider) CollectLogs(dir string, nodes []nodes.Node) error {
 // The info is cached on the first time of the execution.
 func (p *provider) Info() (*providers.ProviderInfo, error) {
 	if p.info == nil {
-		p.info = info(p.logger)
+		var err error
+		p.info, err = info(p.logger)
+		if err != nil {
+			return p.info, err
+		}
 	}
 	return p.info, nil
 }
 
-func info(logger log.Logger) *providers.ProviderInfo {
-	euid := os.Geteuid()
+// podmanInfo corresponds to `podman info --format '{{json .}}'`.
+// The structure is different from `docker info --format '{{json .}}'`,
+// and lacks information about the availability of the cgroup controllers.
+type podmanInfo struct {
+	Host struct {
+		CgroupVersion string `json:"cgroupVersion,omitempty"` // "v2"
+		Security      struct {
+			Rootless bool `json:"rootless,omitempty"`
+		} `json:"security"`
+	} `json:"host"`
+}
+
+func info(logger log.Logger) (*providers.ProviderInfo, error) {
+	const podman = "podman"
+	args := []string{"info", "--format", "{{json .}}"}
```

Comment:
nit: maybe use `json` directly instead of the go template version of json.

Reply:
updated